### PR TITLE
Fix: #23 - JQMIGRATE: 'ready' event is deprecated

### PR DESCRIPTION
**Pull request by jengjeng**
#23 
Fix: - JQMIGRATE: 'ready' event is deprecated.

### CODE REVIEW

**Comment by rmariuzzo** [comment]
Thanks for pointing this @jengjeng! I will think about it before merging as I decided to use the “data-api” with the idea that it could be disabled. I have to think a little about it.

**Comment by rmariuzzo** [comment]
@jengjeng can you change the branch to point to `develop` instead of `master`?

**Comment by jengjeng** [comment]
Hi,
How about if we change the ready event to the custom event then we can detach ready event simply by using ```.off```
eg.
```
$(document).on('checkboxes.data-api.ready', dataApiDomReadyHandler);
$(document).ready(function () {
   $(document).trigger('checkboxes.data-api.ready');
});
```
then
```
$(document).off('checkboxes.data-api.ready');
```

**Comment by rmariuzzo** [comment]
Actually, it doesn't matter for the ready event to be namespaced. It would be a race condition trying to turning off an event that run once on DOM ready...

That said, let just use the bit of code you just provided. But instead of using: `$(document).ready(fn)` use the [recommended](https://api.jquery.com/ready/) one: `$(fn)`.

 > As of jQuery 3.0, only the first syntax is recommended [`$( handler )`]

In the end, we just want something as:

```js
$(dataApiDomReadyHandler);
```

File: dist/jquery.checkboxes-1.2.0.js [hunk 1]
```diff
@@ -208,5 +200,9 @@ function _classCallCheck(instance, Constructor) { if (!(instance instanceof Cons
 
     // Register data-api listeners.
     $(document).on('click.checkboxes.data-api', '[data-toggle^=checkboxes]', dataApiClickHandler);
-    $(document).on('ready.checkboxes.data-api', dataApiDomReadyHandler);
+    $(document).on('checkboxes.data-api.ready', dataApiDomReadyHandler);
+    $(function () {
+        $(document).trigger('checkboxes.data-api.ready');
+        $(document).off('checkboxes.data-api.ready');
+    });
```
Comment (rmariuzzo) [inline dist/jquery.checkboxes-1.2.0.js]: @jengjeng all these lines (203-207) can be reduced to a single:

```js
$(dataApiDomReadyHandler);
```

With that we will be ready to merge.

**Pull request merged**